backend/lib/src/services/msp.rs

```diff
@@ -308,6 +336,35 @@ impl MspService {
             out_of_funds_tick: None,
         })
     }
+
+    pub async fn get_file_from_key(&self, file_key: &str) -> Result<FileDownloadResult, Error> {
+        // Create temp url for download
+        let temp_path = format!("uploads/{}", file_key);
+        let upload_url = format!("{}/{}", self.msp_callback_url, temp_path);
+
+        // Create params
+        let params = jsonrpsee::rpc_params![file_key, &upload_url];
+
+        // Make the RPC call to download file and get metadata
+        let res: Value = self
+            .rpc
+            .call("storagehubclient_saveFileToDisk", params)
+            .await
+            .map_err(|e| Error::BadRequest(e.to_string()))?;
+
+        let rpc_response: RpcResponse = serde_json::from_value(res)
+            .map_err(|e| Error::BadRequest(format!("Failed to parse RPC response: {}", e)))?;
+
+        // Convert location bytes to string
+        let location = String::from_utf8_lossy(&rpc_response.success.location).to_string();
+
+        Ok(FileDownloadResult {
+            file_size: rpc_response.success.file_size,
+            location,
+            fingerprint: rpc_response.success.fingerprint,
+            temp_path: temp_path,
```

Comment:
Ok I see how this works now. I have a small doubt tho, we save the file to a temporary path, which we expect the MSP Node to start uploading the file to our "internal" endpoint, right?

We probably shouldn't expect the file to be written to completely already at this stage, so what happens to the responded `FileStream`? Does it get updated as more data is written to file?

Actually, i the other case, where the RPC returns when the entire file is written, do we want to start streaming only until after the file is completely transfered from Node to Backend?

Reply:
So we are not going to have the file_path until the receiving FileStream finishes